planner/core/common_plans.go

```diff
@@ -416,27 +417,42 @@ func (e *Execute) rebuildRange(p Plan) error {
 	case *PhysicalTableReader:
 		ts := x.TablePlans[0].(*PhysicalTableScan)
 		var pkCol *expression.Column
-		if ts.Table.PKIsHandle {
-			if pkColInfo := ts.Table.GetPkColInfo(); pkColInfo != nil {
-				pkCol = expression.ColInfo2Col(ts.schema.Columns, pkColInfo)
+		if ts.Table.IsCommonHandle {
+			pk := tables.FindPrimaryIndex(ts.Table)
+			pkCols := make([]*expression.Column, 0, len(pk.Columns))
```

Comment:
`pkCols` and `pkColsLen` is available in `AccessPath`, if we add them in PhysicalTableScan, then we don't need to rebuild them for each execution.